include/pqxx/internal/conversions.hxx

```diff
@@ -1156,6 +1156,37 @@ inline constexpr format param_format(std::vector<std::byte, Args...> const &)
 template<typename T> inline constexpr bool is_sql_array<std::vector<T>>{true};
 
 
+#if defined(PQXX_HAVE_SPAN) && __has_include(<span>)
+template<typename T, size_t Extent>
+struct nullness<std::span<T, Extent>> : no_null<std::span<T, Extent>>
+{};
+
+
+template<typename T, size_t Extent>
+struct string_traits<std::span<T, Extent>>
+        : internal::array_string_traits<std::span<T, Extent>>
```

Comment:
I don't think this works with `std::dynamic_extent`!

Reply:
I didn't extend tests, but it should, dynamic_extent is a size_t value

Reply:
Oh, you're right.  It should work — nothing in the template actually uses `Extent` as a size.